src/main/java/org/schemaspy/db/config/PropertiesResolver.java

```
@@ -0,0 +1,90 @@
+package org.schemaspy.db.config;
+
+import org.schemaspy.db.exceptions.RuntimeIOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class PropertiesResolver {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PropertiesResolver.class);
+
+    private final ResourceFinder resourceFinder;
+
+    public PropertiesResolver() {
+        resourceFinder = new PropertiesFinder();
+    }
+
+    public PropertiesResolver(ResourceFinder resourceFinder) {
+        this.resourceFinder = resourceFinder;
+    }
+
+    public Properties getDbProperties(String dbType) {
+        ResolutionInfo resolutionInfo = new ResolutionInfo(dbType);
+        URL url = resourceFinder.find(dbType);
+        resolutionInfo.addTrace(url);
+        Properties props = fromURL(url);
+        processIncludes(props, resolutionInfo);
+        processExtends(props, resolutionInfo);
+        LOG.info(resolutionInfo.getTrace());
+        return props;
```

Maybe if null throw a InvalidConfigurationException?